dbc2val/dbcfeederlib/serverclientwrapper.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+#################################################################################
+# Copyright (c) 2022 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License 2.0 which is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# SPDX-License-Identifier: Apache-2.0
+#################################################################################
+
+import logging
+from typing import Any
+import json
+
+from dbcfeederlib import clientwrapper
+
+
+from kuksa_client import KuksaClientThread
+
+log = logging.getLogger(__name__)
+
+
+class ServerClientWrapper(clientwrapper.ClientWrapper):
+    def __init__(self, ip: str = "localhost", port: int = 8090,
+                 token_path: str = "",
+                 tls: bool = True):
+        # For WebSocket (KUKSA.val Server) we do not send in token information by default
+        # That means in practice that the default defined in kuksa-client will be used
+        # (as of today 'jwt/all-read-write.json.token')
+        log.debug("Initializing ServerClientWrapper")
+        super().__init__(ip, port, token_path, tls)
+
+        # Holder for configs used by kuksa-client
+        # for default values
+        # see https://github.com/eclipse/kuksa.val/blob/master/kuksa-client/kuksa_client/cli_backend/__init__.py
+        self._client_config = {}
+        # Set read-only configs, others to be set just before we use config as they may change
+        self._client_config["protocol"] = "ws"
+        self._kuksa = None
+
+    def get_client_specific_configs(self):
+        """
+        Get client specific configs and env variables
+        """
+        log.debug("No additional configs for KUKSA.val server")
+
+    def start(self):
+        """
+        Start connection to server and authorize
+        """
+
+        # Finalize config
+        self._client_config["ip"] = self._ip
+        self._client_config["port"] = self._port
+        self._client_config["insecure"] = not self._tls
+        # Do not set token if it is empty to allow default client lib info to be used
+        if self._token_path != "":
+            self._client_config["token"] = self._token_path
+
+        # TODO add data for root cert if using TLS and if given
+
+        self._kuksa = KuksaClientThread(self._client_config)
```

Comment:
do we want to us threaded API here?

Reply:
This file is for KUKSA.val server (i.e. websocket) and has not been changed in this PR - the server-specific code has just been put in separate file to make `dbcfeeder.py` unaware of what APIs that actually are used. What other options do we have?

Reply:
Ah okay, I missed that. fought it was in the databrokerclientwrapper. Then its fine